### PR TITLE
Fix various issues

### PR DESCRIPTION
**Pull request by AshleyYakeley**
* Stop parser from hanging and eating up memory
* Remove timeout mechanism entirely
* Fix all build warnings
* Follow the dconf dump format more accurately
* Support variants and records

There might still be output issues.

### CODE REVIEW

**Comment by AshleyYakeley** [comment]
I think the backslash character literal parsing code may be wrong. I'm not sure how it's supposed to work.

**Comment by gvolpe** [comment]
Thanks for doing this! Can you also update [this](https://github.com/gvolpe/dconf2nix/blob/master/.github/workflows/ci.yml#L11) to `ubuntu-latest` so this can be verified on CI?

**Comment by AshleyYakeley** [comment]
OK done

**Comment by gvolpe** [comment]
Need to bump the version of all actions to the latest, as this project hasn't seen an update in a while.

**Comment by AshleyYakeley** [comment]
OK, I think you'll have to do that?

**Comment by gvolpe** [comment]
You can do it on this PR in the same way you updated the `ubuntu-latest` value.

**Comment by AshleyYakeley** [comment]
The CI is broken in master, is it not? Fix it in master and then I'll merge it in.

**Comment by jtojnar** [comment]
Opened CI fix pr https://github.com/gvolpe/dconf2nix/pull/81

File: output/dconf.nix [hunk 1]
```diff
@@ -15,13 +15,13 @@ with lib.hm.gvariant;
 
     "org/gnome/desktop/interface" = {
       clock-show-weekday = true;
-      document-font-name = "JetBrainsMono Nerd Font 11";
+      document-font-name = " JetBrainsMono Nerd Font 11 ";
```
[the remark this answers is not in the document]
Comment (AshleyYakeley) [reply output/dconf.nix:18]: These spaces are present in the test input file. Here's the input line that causes this particular output:

```
document-font-name=' JetBrainsMono Nerd Font 11 '
```

I can go through and change them if you want.

[the remark this answers is not in the document]
Comment (AshleyYakeley) [reply output/dconf.nix:18]: The previous parser stripped the spaces out of the quoted strings, whereas my parser in this PR does not.

**Review by gvolpe — APPROVED** [review]
Thank you so much, this is pretty cool!

File: output/dconf.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (gvolpe) [reply output/dconf.nix:18]: Fine with me as it is now 👍🏽

**Pull request merged**